docs/proposals/componentconfig-dump.md

```diff
@@ -0,0 +1,101 @@
+* helping complete the migration of flags to componentconfigs
+
+## Background
+
+The document is built from SIG-Scale and PR discussions. Here we document the results of these and build the proposal of what changes should/will be made to K8s.
+
+## Motivation
+
+By necessity Kubernetes is a stateful thing made up of components who have been configured in a particular way. Currently the K8s code base, as 09/2016, does not support a way nor means by which to query for the set of configurations that led to the current state of any component. As a consequence it is extremely difficult to debug the current state of a K8s cluster, especially when original configurations have been lost, or are simply not available. 
+
+## Design
+
+Components will have either a path indicated by annotations that can point to how to retrieve the componentconfigs or through a well known component api path, as discussed below.
+
+### Component API changes
+
+Components (like kubelet) have endpoints that can be altered to include an endpoint such as
+
+```
+ /componentconfigs 
+```
+
+similar to /metrics endpoints for K8s components and likely to be authenticated in much the same way.
+
+By way of an HTTP GET the output would be:
+
+```
+{
+    "kind": "componentconfigs",
+    "apiVersion": "alpha",
+    "metadata": {
+        "component": "kubelet",
+        "id": "818B908B-D053-CB11-BC8B-EEA826EBA090"
+        "timeStamp": "2016-09-13T14:47:35+00:00",
+    },
+    "configs": {
+        "cloud-provider": "aws",
+            ...
+    }
+}
+```
+
+Those componentconfigs that are to be associated with a particular component will be annotated so as to be labeled for collection and served by the proposed endpoint. 
+
+## Alternatives and Further Discussions
+
+Here we mention points of contentions and alternatives as we make progress in the proposal.
+
+* Security/Auth has been brought up as a point that strongly affects the proposed endpoint. Ideas around component endpoints and auth have been discussed for `/metrics` and it has been noted that SIG-Auth may be assigned the `/componentconfigs` endpoint to ensure auth is handled correctly.
+* The addition of an api change/addition to APIServer has been discussed, and for reasons of security it was agreed to be left out as a potential addition in the future.
+* The addition of kubectl/kubeadmin interactions has also been scrapped for reasons of security with possible addition in the future, but do not gain us much for our present motivations.
+* It has been agreed that extracting configs from componentconfigs is the way forward.
+
+## Future Work
+
+We discuss here future work that should be considered an extension of this proposal, and may become actionable work to be included for present development. The idea here is to extend our endpoint with the additional `PUT HTTP method`, as a way to configure components, the motivation is as follows:
```

Comment:
@bgrant0607 absolutely correct. Meant to change this since the components themselves are not directly in contact with etcd and their state info is gathered from apiserver interactions. Do you have any idea if the plans for the future for splitting certain components to be standalone or not? Also how opposed would you be about opening the apiserver handling such an endpoint?